### ISSUE TITLE
Hyperparameters in paper vs code

### ISSUE THREAD

**Issue by tfy14esa** [YvanYin/VNL_Monocular_Depth_Prediction#35]
Hi,
Thanks for releasing this code! I think the idea you present is very neat. I have a few questions:
1. I understand that the network outputs a depth between 0-1 which needs to be scaled properly to get the metric depth, but do you apply the loss function on the scaled ground truth depth images (i.e. between 0-1) or do you apply the loss function in the metric space? I ask this because I need to know if the hyperparameters are tuned in the metric space or in the scaled space between 0-1. From the paper, it appears that you apply the loss in the metric space e.g. because you have one hyperparameter theta = 0.6 m in the paper. I would appreciate if you can confirm this.
2. My second question concerns trying to understand how the names of the hyperparameters in the code are related to the names in the paper. Specifically what are the following hyperparameters called in the code?:
theta = 0.6 m
alpha = 120 degrees
beta = 30 degrees
Please let me know if I missed any hyperparameter from the paper.
Furthermore, I do not understand the following hyperparameters in the code (see the class VNL_Loss and function filter_mask):
delta_cos = 0.867
delta_diff_x = 0.005
delta_diff_y = 0.005
delta_diff_y = 0.005
delta_z = 0.0001

I could not properly grasp how these hyperparameters relate to the ones you discuss in the paper. I would be very grateful if you can educate me on this.

Thanks in advance!
Cheers,
Erik

**Comment by YvanYin** [comment]
1. In the scaled depth space, I have tried it. It can also work. Please refer to the paper 'DiverseDepth: Affine-invariant Depth Prediction Using Diverse Data.' (github [https://github.com/YvanYin/DiverseDepth]) .
2. delta_diff_x/y/z is the distance difference between 2 points along x/y/z axis. delta_cos is the cosine is the `cos<AB, AC>`.   delta_z is the threshold for the valid depth valid. z > delta_z is valid.

**Comment by tfy14esa** [comment]
Thank you very much for your quick reply. 
1. So this means that the variables gt_depth and pred_depth are assumed to be in the range 0-1, right? And this, in turn, means that the variables delta_z, delta_diff_x/y/z are also in this space and if I were to use this loss in the metric space e.g. let's say I assume that gt_depth and pred_depth are in the range 0-10 meters, I would want to multiply delta_z, delta_diff_x/y/z with 10, right?
2. So to confirm, according to the paper, you constrain the L2 norm between sampled points to be at least 0.6 m, but in the code you instead check that along each axis the distance is larger than delta_diff_x/y/z? Just as a sanity check: you use delta_diff_x/y/z = 0.005. If we transform this hyperparameter into the metric space 0-10m according to the reasoning in "1.", we get delta_diff_x/y/z = 0.05 m. This value does not reflect the theta= 0.6 m that is reported in the paper and 0.05 m seem like a very low number to me. Could you perhaps educate me on why you select such a low number for delta_diff_x/y/z? Also, you say delta_cos is the angle restriction of cos<AB, AC> i.e. cos<AB, AC> needs to be less than 0.867 = cos(30 degrees), but more than -0.867 = cos(150 degrees). So alpha = 150 degrees in the code, correct?

Again, thanks for your kind reply!
Cheers,
Erik

**Comment by tfy14esa** [comment]
Hi YvanYin,
I just wanted to send you a reminder here, in case my previous message got lost. Take your time though, no rush. 

Best,
Erik

**Comment by YvanYin** [comment]
Hi
1) In the training, the gt_depth is normalized to 0-1. If the gt_depth is in the 0-10m range, you have to adjust delta_z, delta_diif_x/y/z.
2) In our naive implementation, we set the distance to 0.6m. In the released code, we set the delta_diiff_x/y/z to make sampled points far away from each other. You can also set the distance between them, but these two methods are the same. We didn't do the ablation study to compare them. About the alpha, the angle between <ab, ac> is [30, 150].
